README.md

```diff
@@ -90,7 +90,7 @@ fn sum_as_string(a: usize, b: usize) -> PyResult<String> {
 /// import the module.
 #[pymodule]
 fn string_sum(m: &Bound<'_, PyModule>) -> PyResult<()> {
-    m.add_function(wrap_pyfunction!(sum_as_string, m)?)?;
+    PyModule::add_function(m, wrap_pyfunction!(sum_as_string, m)?)?;
```

Comment:
There's a lot of these changes to use path syntax, why?

Reply:
These were the way I found to resolve errors like:

```
error[E0599]: no method named `as_type_ptr` found for reference `&pyo3::Bound<'_, PyType>` in the current scope
 --> tests/test_field_cfg.rs:5:1
  |
5 | #[pyclass]
  | ^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pyclass` (in Nightly builds, run with -Z macro-backtrace for more info)
help: there is a method `as_ptr` with a similar name
  |
5 | as_ptr
  |
```

Reply:
If you want to play around with this, checkout https://github.com/PyO3/pyo3/pull/4885/commits/c8a6ebe334a57bdb4f00b2ecf968adb75dbea375.

Reply:
I've added a comment about this upstream: https://github.com/rust-lang/rust/pull/135881#issuecomment-2647844621

Reply:
The suggestion from upstream solved this completely.